ocaml/libs/tracing/tracing.ml

```diff
@@ -220,7 +225,8 @@ module Spans = struct
               Hashtbl.replace spans key filtered_list ;
               match Hashtbl.find_opt finished_spans key with
               | None ->
-                  Hashtbl.add finished_spans key [span]
+                  if Hashtbl.length spans < !max_traces then
```

Comment:
copy+past error here, did you mean `finished_spans` instead of `spans`?

Reply:
[Resolved ](https://github.com/xapi-project/xen-api/pull/4958/commits/34e9795b2b9c0e4964af7369f7f4b515105c3ea4)